File: packages/create-react-native-library/templates/java-view-mixed-library/android/src/main/java/com/{%- project.package %}/{%- project.name %}ModuleImpl.java
package com.<%- project.package -%>;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactMethod;

/**
 * This is where the module implementation lives
 * The exposed methods can be defined in the `turbo` and `legacy` folders
 */
public class <%- project.name -%>ModuleImpl  {
  public static final String NAME = "<%- project.name -%>";

  public static void multiply(double a, double b, Promise promise) {
    promise.resolve(a * b);
  }
}

File: packages/create-react-native-library/templates/kotlin-library-mixed/android/src/main/java/com/{%- project.package_dir %}/{%- project.name %}ModuleImpl.kt
package com.<%- project.package -%>;

import com.facebook.react.bridge.Promise

/**
 * This is where the module implementation lives
 * The exposed methods can be defined in the `new` and `legacy` folders
 */
object <%- project.name -%>ModuleImpl {
  const val NAME = "<%- project.name -%>"

  fun multiply(a: Double, b: Double, promise: Promise) {
    promise.resolve(a * b)
  }
}
